core/src/main/java/org/elasticsearch/common/blobstore/fs/FsBlobContainer.java

```diff
@@ -85,7 +86,13 @@ public FsBlobContainer(FsBlobStore blobStore, BlobPath blobPath, Path path) {
     @Override
     public void deleteBlob(String blobName) throws IOException {
         Path blobPath = path.resolve(blobName);
-        Files.deleteIfExists(blobPath);
+        try {
+            if (!Files.deleteIfExists(blobPath)) {
+                throw new IOException("File " + blobPath.toString() + " does not exist");
+            }
+        } catch (DirectoryNotEmptyException | SecurityException e) {
+            throw new IOException(e);
```

Comment:
We should not catch the `SecurityException` at all.  Let it propagate.  We should not have even gotten to this point if the security manager did not give us access here, but in any case, its not an exception we should handle at this level.  It should just be propagated.